rtl/include/eclhelper.hpp

```diff
@@ -727,6 +727,7 @@ interface ICodeContext : public IResourceContext
     virtual char *getPlatform() = 0; // caller frees return string.
     virtual unsigned getPriority() const = 0;
     virtual char *getWuid() = 0; // caller frees return string.
+    virtual unsigned getWorkflowId() const = 0; // Note: don't use yet as it has not been fully implemented in all derived classes
```

Comment:
I think you need to place any new virtuals at end of interface.
This will allow existing queries compiled with previous version to continue to work without recompilation, if methods are placed in the middle, the VMT offsets of methods past it will change and cause problems.

Reply:
Yes, it needs to be at the end of the interface, and it cannot match the name of an existing virtual either.